temporaltest/server.go

```diff
@@ -138,3 +138,47 @@ func NewServer(opts ...TestServerOption) *TestServer {
 
 	return &ts
 }
+
+// NewServerWithTls starts and returns a new TestServer.
+//
+// If not specifying the WithT option, the caller should execute Stop when finished to close
+// the server and release resources.
+func NewServerWithTls(caCertificate, certificate, key string, useMtls bool, defaultClientOptions client.Options, opts ...TestServerOption) *TestServer {
```

Comment:
I implemented it the way I did because I was hesitant about the scope of those changes. It turned it out to be smaller than I expected, and it is in now.